### PR TITLE
Make expander arrows in the JSON editor red if one of their descendants has a problem

### PR DESCRIPTION
**Pull request by rogerl50**
Currently it can be hard to figure out if there was an error in a complicated routine call, as only the erroneous statement shows up red. This PR bubbles the error up, so that the expandable arrow on any node will be red if one of its descendants had an error.

### CODE REVIEW

**Comment by bjalder26** [comment]
This looks like it works to me.  Also, this will be a really nice feature.
image

**Comment by bjalder26** [comment]
> > This looks like it works to me. Also, this will be a really nice feature.
> 
> What room is this in?

http://212.47.248.129:4040/Interitus

**Comment by rogerl50** [comment]
> > What room is this in?
> 
> http://212.47.248.129:4040/Interitus
> 
> Those errors are intentional.

I added some more code, some correct and some with errors, and the coloring still appears correct.

File: client/js/jsonedit.js [hunk 1]
```diff
@@ -1356,14 +1356,27 @@ function jeLoggingRoutineEnd(variables, collections) {
   --jeLoggingDepth;
   if(!jeLoggingDepth) {
     $('#jeLog').innerHTML = jeLoggingHTML + '</div></div>';
-    var expanders = document.getElementsByClassName('jeExpander');
-    var i;
+    // Make it so clicking on the arrows expands the subtree
+    const expanders = document.getElementsByClassName('jeExpander');
```
Comment (ArnoldSmith86) [inline client/js/jsonedit.js:1360]: `$a('.jeExpander')` should do the same. Yours might be slightly faster so just FYI.

**Review by ArnoldSmith86 — APPROVED** [review]
Code looks good to me.

Feature looks awesome to me.

File: client/js/jsonedit.js (hunk 1 above)
Comment (rogerl50) [reply client/js/jsonedit.js:1360]: Do you mean `$('.jeExpander')`?

Comment (ArnoldSmith86) [reply client/js/jsonedit.js:1360]: No. That only returns the first one. `a` stands for `all` (or `array`) and returns all of them in an array.

Comment (rogerl50) [reply client/js/jsonedit.js:1360]: Good to know. Thanks. (Where is this documented in, say, the w3schools pages?)

Comment (ArnoldSmith86) [reply client/js/jsonedit.js:1360]: Nope: https://github.com/ArnoldSmith86/virtualtabletop/blob/main/client/js/domhelpers.js#L1-L7

Comment (rogerl50) [reply client/js/jsonedit.js:1360]: Oh. I never looked at that file. Guess I should :)

**Pull request merged**